network/framework/src/protocols/network/mod.rs

```diff
@@ -394,13 +394,19 @@ impl<TMessage: Message> NetworkSender<TMessage> {
         req_msg: TMessage,
         timeout: Duration,
     ) -> Result<TMessage, RpcError> {
-        // serialize request
-        let req_data = protocol.to_bytes(&req_msg)?.into();
+        // Serialize the request using a blocking task
+        let req_data = tokio::task::spawn_blocking(move || protocol.to_bytes(&req_msg))
```

Comment:
presumably this tested to make things faster, but I dislike the complexity and overhead it adds